lexicon/providers/auto.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import absolute_import
+
+import subprocess
+import re
+import os
+import pkgutil
+import logging
+import argparse
+import importlib
+import argparse
+import re
+
+from builtins import object
+
+import six
+import tldextract
+import sys
+
+from lexicon import providers
+from lexicon.common.options_handler import env_auth_options
+
+LOGGER = logging.getLogger(__name__)
```

Comment:
just a nit. can we change this to `logger` like we use in the other classes?

Reply:
Of course. Next commit.